server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -126,6 +138,25 @@ public Builder(String name, NumberType type, ScriptCompiler compiler, boolean ig
                     }
                 });
 
+            this.metric = TimeSeriesParams.metricParam(
+                m -> toType(m).metricType != null ? toType(m).metricType.name() : null,
+                null,
+                MetricType.gauge.name(),
+                MetricType.counter.name()
+            );
+            // We are overriding the default validator that checks for acceptable values.
+            // Therefore, we must call the default validator explicitly from inside our new validator.
+            // TODO: Make parameters support multiple validators
+            final Consumer<String> defaultValidator = metric.getValidator();
+            metric.setValidator(v -> {
+                // Call the default validator first
+                defaultValidator.accept(v);
```

Comment:
@romseygeek do you think this should be easier? Or the default? Or something?

Reply:
Maybe add a `wrapValidator` method that does this for you?

Reply:
I was thinking that perhaps we could add a list of validators for a parameter. Those validators can be executed sequentially. In this case a parameter can have its validator and its subclasses can add more.

It's an easy change and I can submit a different PR for that. WDYT @romseygeek?

Reply:
Sounds good to me.